src/Transports/MassTransit.AmazonSqsTransport/Pipeline/AmazonSqsMessageReceiver.cs

```diff
@@ -114,11 +118,11 @@ async Task PollMessages(ChannelExecutor executor)
             var receiveCount = 1;
             var messageLimit = Math.Min(_receiveSettings.PrefetchCount, 10);
 
-            while (!IsStopping)
+            while (!_cancellationTokenSource.IsCancellationRequested)

```

Comment:
This is absolutely pointless, the Agent already has a cancellation Token.

Reply:
make sense here, but for client request we should carry other token, which might be cancelled during execution